ztp/gitops-subscriptions/argocd/SecureBoot.md

```diff
@@ -0,0 +1,80 @@
+# Enable and Verify for Secure Boot for SNO
+
+## How to Enable Secure Boot 
+
+ - Configure `SiteConfig` with `UEFIScureBoot` for `bootMode`. E.g part of yaml below 
+ ```yaml
+nodes:
+  - hostName: "myhost"
+    bootMode: "UEFISecureBoot"
+```
+
+- This value should now be available in the Hub (cluster w/ ACM) cluster's appropriate BareMetalHost (bmh) CR. e.g part of the yaml below 
+```yaml
+spec:
+  bootMode: UEFISecureBoot
+```
+
+## How Verify Secure Boot Status
+
+Right now the installation verification is done by `ssh`-ing into the node. Credentials such as private key associated with the node are required.
+E.g `ssh` command 
+```shell
+ssh -i key core@myhost.rh.com
+```
+
+Once logged into the node there are a couple of ways to verify. 
+  - **With `journalclt`**
+```shell
+[core@myhost ~]$ journalctl | grep secureboot
+Feb 25 01:25:02 localhost kernel: secureboot: Secure boot enabled
+```
+
+  - **With `mokutil`**
+```shell
+mokutil --sb-state
+SecureBoot enabled
+```
+
+`motutil` may not work correctly and see something like this below
+```shell
+mokutil --sb-state
+EFI variables are not supported on this system
+```
+
+To use `mokutil` do the following: 
+
+  - Create MachineConfig CR `99-efi-runtime-path-kargs.yaml`
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 99-efi-runtime-path-kargs
+spec:
+  kernelArguments:
+  - "efi=runtime"
```

Comment:
Maybe it would be better to add this to the PAO performance profile where we enable the RT kernel, since it's only needed if you have the RT kernel installed anyway...

Reply:
So appending to [this](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/source-crs/PerformanceProfile.yaml#L8) list? That would mean the user would enable the runtime regardless of using SecureBoot right? Is that what we would want now?

Reply:
I don't think there's any harm in enabling this, though I think we'd have to be careful and check what happens if you add this entry to the non-realtime kernel.   But it's much more a feature of the RT kernel and whether it has access to any EFI variables, regardless of whether or not secure boot is enabled.

Reply:
To put it another way, if you install the RT kernel and don't add this kernel argument, and try to use mokutils to verify that secureboot is NOT enabled, you would have the same error regardless of whether secureboot is enabled on not enabled!

Reply:
Can confirm adding `efi=runitme` to `PerformanceProfile` and setting RT Kernel to false, mokutil works as expected 

```
[core@cnfdf04 ~]$ uname -r     
4.18.0-305.34.2.el8_4.x86_64
[core@cnfdf04 ~]$  mokutil --sb-state
SecureBoot enabled

```

Reply:
Excellent!